ckanext/validation/logic.py

```diff
@@ -629,25 +627,32 @@ def resource_create(up_func, context, data_dict):
 
     # Custom code starts
 
-    run_validation = True
+    if get_create_mode_from_config() == 'sync':
 
-    for plugin in plugins.PluginImplementations(IDataValidation):
-        if not plugin.can_validate(context, data_dict):
-            log.debug('Skipping validation for resource {}'.format(resource_id))
-            run_validation = False
+        run_validation = True
 
-    if run_validation:
-        is_local_upload = (
-            hasattr(upload, 'filename') and
-            upload.filename is not None and
-            isinstance(upload, uploader.ResourceUpload))
-        _run_sync_validation(
-            resource_id, local_upload=is_local_upload, new_resource=True)
+        for plugin in plugins.PluginImplementations(IDataValidation):
+            if not plugin.can_validate(context, data_dict):
```

Comment:
Why skip validation if one plugin can do it and another can't? Shouldn't we just let the one plugin handle it?